runtimeconfig/manager.go

```diff
@@ -180,7 +208,10 @@ func (om *Manager) loadConfig() error {
 
 	// expose hash of runtime config
 	om.configHash.Reset()
-	om.configHash.WithLabelValues(fmt.Sprintf("%x", hash[:])).Set(1)
+	om.configHash.WithLabelValues(fmt.Sprintf("%x", hash)).Set(1)
+
+	// preserve hashes for next loop
+	om.fileHashes = hashes
```

Comment:
The other updates here (`om.setConfig`) take the mutex to update. The hashes need to take the mutex to update and while being read in the loop above.

Reply:
`om.fileHashes` doesn't need extra synchronization, as commented on the field. It's only accessed from `loadConfig`, which is called from `Starting` or `Running` state (called `loop`) functions of the service, and these two states already form happens-before.